src/Blocks/BlockEnchantingTable.h

```diff
@@ -0,0 +1,95 @@
+
+#pragma once
+
+#include "BlockHandler.h"
+#include "../Item.h"
+#include "../UI/EnchantingWindow.h"
+#include "../BlockEntities/EnchantingTableEntity.h"
+#include "../Entities/Player.h"
+
+
+
+
+
+class cBlockEnchantingTableHandler :
+	public cBlockHandler
+{
+	using Super = cBlockHandler;
+
+public:
+	cBlockEnchantingTableHandler(BLOCKTYPE a_BlockType):
+		Super(a_BlockType)
+	{
+	}
+
+
+
+
+
+	virtual bool OnUse(
+		cChunkInterface & a_ChunkInterface,
+		cWorldInterface & a_WorldInterface,
+		cPlayer & a_Player,
+		const Vector3i a_BlockPos,
+		eBlockFace a_BlockFace,
+		const Vector3i a_CursorPos
+	) override
+	{
+		AString WindowName = "Enchant";
+		a_WorldInterface.DoWithBlockEntityAt(a_BlockPos.x, a_BlockPos.y, a_BlockPos.z, [&] (cBlockEntity & a_Entity)
+		{
+			if (a_Entity.GetBlockType() != E_BLOCK_ENCHANTMENT_TABLE)
+			{
+				return false;
+			}
+			auto EnchantingTable = static_cast<cEnchantingTableEntity &>(a_Entity);
+			if (!EnchantingTable.GetCustomName().empty())
+			{
+				WindowName = EnchantingTable.GetCustomName();
+			}
```

Comment:
You should use a temporary variable for the custom name here, to avoid violating DRY and prevent any issues with mistyping and later maintenance (although this looks correct).

Reply:
You mean not defaulting to Enchant but asking fot the name and checking if there is a custom name?

Reply:
I was just referring to a very minor point, I intended that the above code would become:

```cpp
auto CustomName = EnchantingName.GetCustomName();
if (!CustomName.empty())
{
    WindowName = CustomName;
}
```

That way you avoid calling GetCustomName twice, and potentially causing issues later when one is changed but not the other; or if one is mistyped etc.

Reply:
Oh - now I got it  - added this